### ISSUE TITLE
Wrapper types for serializing `Cow`s

### ISSUE THREAD

**Issue by djkoloski** [djkoloski/rkyv#172]
`Cow`s have lifetime semantics that aren't compatible with serializing and deserializing by default. We can provide some wrapper types that make it easier to work with them though:

- `AsOwned`: Serializes and deserializes each `Cow` as if it were owned.
- `AsShared`: Serializes each `Cow` as if it were shared, deserializes as if it were owned (due to lifetime semantics).

It might be good to look for crates that offer a `Cow`+`Rc`-like functionality and get implementations for them. Without the lifetime, it would be possible to provide good default implementations for `Serialize` and `Deserialize` similarly to how `Rc` already works.

**Comment by PonasKovas** [comment]
```rust
pub struct AsOwned;

impl<'a, T: 'a + ToOwned + ?Sized + ArchiveUnsized> ArchiveWith<Cow<'a, T>> for AsOwned {
    type Archived = rkyv::boxed::ArchivedBox<<T as ArchiveUnsized>::Archived>;
    type Resolver = rkyv::boxed::BoxResolver<<T as ArchiveUnsized>::MetadataResolver>;

    unsafe fn resolve_with(
        field: &Cow<'a, T>,
        pos: usize,
        resolver: Self::Resolver,
        out: *mut Self::Archived,
    ) {
        rkyv::boxed::ArchivedBox::resolve_from_ref(field.as_ref(), pos, resolver, out);
    }
}
impl<'a, T: 'a + ToOwned + ?Sized, S: Fallible + ?Sized> SerializeWith<Cow<'a, T>, S> for AsOwned
where
    T: SerializeUnsized<S>,
{
    fn serialize_with(field: &Cow<'a, T>, serializer: &mut S) -> Result<Self::Resolver, S::Error> {
        rkyv::boxed::ArchivedBox::serialize_from_ref(field.as_ref(), serializer)
    }
}
impl<'a, T, D> DeserializeWith<rkyv::boxed::ArchivedBox<T::Archived>, Cow<'a, T>, D> for AsOwned
where
    T: 'a + ToOwned + ArchiveUnsized + ?Sized,
    T::Archived: DeserializeUnsized<T, D>,
    D: Fallible + ?Sized,
{
    fn deserialize_with(
        field: &rkyv::boxed::ArchivedBox<T::Archived>,
        deserializer: &mut D,
    ) -> Result<Cow<'a, T>, D::Error> {
        Ok(Cow::Owned(
            Deserialize::<Box<T>, D>::deserialize(field, deserializer)?.to_owned(),
        ))
    }
}
```

this is what I have

**Comment by djkoloski** [comment]
Moving this to a minor release.

**Comment by djkoloski** [comment]
Fixed by a773ba2c55390ab76b269985dc2009bb1a775253
